melosys-eessi-app/src/main/java/no/nav/melosys/eessi/service/sed/JsonFieldMasker.kt

```diff
@@ -0,0 +1,102 @@
+                return if (node.isNumber) {
+                    // Mask numbers as their size-based numeric value
+                    val size = node.asText().length
+                    val maskedNumber = (1..size).joinToString("") { it.toString() }.toInt()
+                    LongNode(maskedNumber.toLong()) // Ensure it's treated as a number
+                } else {
+                    // Mask text values as 'xxxx...'
+                    TextNode("x".repeat(node.asText().length))
+                }
+            }
+            fun sanitizeArrayNode(node: JsonNode, keep: Set<String>): ArrayNode {
+                val sanitizedArray = node.deepCopy<ArrayNode>()
+                node.forEachIndexed { index, element ->
+                    val sanitizedElement: JsonNode = when {
+                        element.isTextual -> maskPrimitive(element)
+                        element.isInt || element.isLong -> maskPrimitive(element)
+                        else -> sanitizeNode(element, keep) // Recursively sanitize
+                    }
+                    sanitizedArray.set(index, sanitizedElement) // Ensure sanitizedElement is JsonNode
+                }
+                return sanitizedArray
+            }
+
+            fun sanitizeObjectNode(node: JsonNode, keep: Set<String>): ObjectNode {
+                val sanitizedObject = node.deepCopy<ObjectNode>()
+                node.fieldNames().forEachRemaining { field ->
+                    val fieldValue = node[field]
+                    if (keep.contains(field)) {
+                        sanitizedObject.set<JsonNode>(field, fieldValue) // Preserve whitelisted fields
+                    } else {
+                        // Mask text and numbers, sanitize nested objects and arrays
+                        sanitizedObject.set<JsonNode>(
+                            field, when {
+                                fieldValue.isTextual -> maskPrimitive(fieldValue)
+                                fieldValue.isInt || fieldValue.isLong -> maskPrimitive(fieldValue)
+                                else -> sanitizeNode(fieldValue, keep)
+                            }
+                        )
+                    }
+                }
+                return sanitizedObject
+            }
+
+            return when {
+                node.isObject -> sanitizeObjectNode(node, keep)
+                node.isArray -> sanitizeArrayNode(node, keep)
+                node.isBoolean -> node // Preserve boolean values
+                node.isInt || node.isLong -> TextNode("n".repeat(node.asText().length)) // Mask integer/long values
+                else -> node // Return other types (e.g., null, floats) as-is
+            }
+        }
+
+        return sanitizeNode(mapper.readTree(json), whitelist).toPrettyString()
+    }
+
+    fun toMaskedJson(sedDataDto: SedDataDto): String =
+        toMaskedJson(
+            sedDataDto, setOf(
+                "sedType", "kjoenn", "adressetype", "land", "relasjon",
```

Comment:
Tror ikke vi ønsker og trenger `kjoenn` og `statsborgerskap`, eller noen opplysninger om bruker i det hele tatt.
`relasjon` i `FamilieMedlem` er heller ikke relevant

Reply:
Jo mer vi har som kan være til hjelp for å reprodusere feilen i en test jo raskere får vi løst feil. 
Tenker statsborgerskap kan være nyttig siden det treffer noe logik. 
Men relasjon kan vi nok fjerne ja